contracts/WBTCFlashLoan.sol

```diff
@@ -0,0 +1,212 @@
+    IERC20 public wbtc = IERC20(0x2260FAC5E5542a773Aa44fBCfeDf7C193bc2C599);
+    IWETH public weth = IWETH(0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2);
+    CErc20I public cToken = CErc20I(0xccF4429DB6322D5C611ee964527D42E5d685DD6a);
+    CEtherI public cEth = CEtherI(0x4Ddc2D193948926D02f9B1fE9e1daa0718270ED5);
+    IUniswapAnchoredView public oracle = IUniswapAnchoredView(0x841616a5CBA946CF415Efe8a326A621A794D0f97);
+    uint constant public PRICE_DECIMALS = 10 ** 6;
+
+    uint256 constant public collateralRatioETH = 0.75 ether; // 75%
+    uint marketId = 0; // ETH
+
+    event Number(string name, uint number);
+
+    // needs to be able to receive ETH
+    receive() external payable {
+
+    }
+
+    function takeWBTCFlashLoan() public {
+        uint256 desiredWBTC = 1e8; // 1 WBTC
+        // calculate how much ETH we need to get WBTC
+        // price ETH / BTC
+        uint ETHBTC = oracle.price("ETH").mul(PRICE_DECIMALS).div(oracle.price("BTC"));
+        // ETH = desiredWBTCinETH / collatRatioETH (remember to adjust decimals 8 => 18)
+        uint256 amountRequiredETH = 
+            desiredWBTC.mul(PRICE_DECIMALS).mul(1e10)
+                .mul(1e18).div(collateralRatioETH)
+                .div(ETHBTC);
+
+        emit Number("ETHUSD", oracle.price("ETH"));
+        emit Number("BTCUSD", oracle.price("BTC"));
+        emit Number("ETHBTC", ETHBTC);
+        emit Number("amountRequiredETH", amountRequiredETH);
+
+        amountRequiredETH = amountRequiredETH.mul(105).div(100); // 5% more just in case
+        emit Number("amountRequiredETH", amountRequiredETH);
+
+        // uint256 dydxLiquidity = _getDyDxLiquidity();
+        uint256 dydxLiquidity = 1000 ether;
+
+        // adjust for liquidity 
+        amountRequiredETH = Math.min(amountRequiredETH, dydxLiquidity);
+
+        Actions.ActionArgs[] memory actions = new Actions.ActionArgs[](3);
+
+        // take ETH flash loan
+        actions[0] = _getWithdrawAction(amountRequiredETH);
+
+        // CALL OPERATE
+            // supply ETH to Compound
+            // borrow desired WBTC from Compound
+            // do stuff with WBTC
+            // repay desired WBTC to Compound
+            // withdraw ETH from Compound
+        actions[1] = _getCallAction(desiredWBTC);
+
+        // repay ETH flash loan
+        actions[2] = _getDepositAction(amountRequiredETH.add(2));
+
+        Account.Info[] memory accounts = new Account.Info[](1);
+        accounts[0] = Account.Info({
+            owner: address(this),
+            number: 1 // why 1?
+        });
+
+        // execute
+        solo.operate(accounts, actions);
+
+        // return to initial balance
+        require(wbtc.balanceOf(address(this)) == 0);
+        require(address(this).balance == 0);
+    }
+
+    function callFunction(
+        address sender,
+        Account.Info memory accountInfo,
+        bytes memory data
+    ) external {
+        require(sender == address(this));
```

Comment:
Critical issue, requires msg.sender to be SOLO

Reply:
this contract was not supposed to be here. deleted

thanks